### DISCUSSION TITLE
How to make Prisma.[ModelName]GetPayload contains computed fields from Prisma Client extensions?

### DISCUSSION THREAD

**Discussion by tithanayut** [prisma/prisma#22134] [Q&A] [answered]
### Question

Suppose we follow [add custom fields or methods to query results](https://www.prisma.io/docs/concepts/components/prisma-client/client-extensions/result#add-custom-fields-or-methods-to-query-results) to use Prisma Client extensions to add computed fields to query results.

```ts
const prisma = new PrismaClient().$extends({
  result: {
    user: {
      myComputedField: {
        needs: { ... },
        compute() { 
          return 'some value'
        }
      },
    },
  },
});
```

Is there a native way to make Prisma.UserGetPayload contains `myComputedField` in the type?

Thank you for your help.

**Comment by ludralph** [comment, marked as answer]
HI @tithanayut 👋 

Thank you for raising this question.

Prisma does not yet natively support computed fields in the type system. The computed fields are added to the result object and are not part of the Prisma-generated types like `Prisma.UserGetPayload`. We use Prisma Client extensions as a way to add computed fields at runtime. To see this specific support, can you please add your usecase to this open [issue](https://github.com/prisma/prisma/issues/3394).

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by tithanayut** [reply]
I see. Thank you!

**Comment by tomkennedy22** [comment]
I added a comment in #3394 inquiring about this - I'm very interested in this functionality

**Comment by oec-fukagawa** [comment]
Hello, I share my workaround.

```typescript
const myExtension = Prisma.defineExtension({
  result: {
    foo: {
      model: {
        compute() {
          return 'Foo' as const;
        },
      },
    },
  }
});

// This function is defined only to extract the type (ExPrisma).
const extensionTypeGenerator = () => new PrismaClient().$extends(myExtension);
export type ExPrisma = ReturnType<typeof extensionTypeGenerator>;

const sharedFooArgs = { ...some complex args... } as const satisfies Prisma.FooDefaultArgs;

// This function also just extracts the type.
const fooTypeGenerator = (p: ExPrisma) => p.foo.findMany(sharedFooArgs);
type Foo = Awaited<ReturnType<typeof fooTypeGenerator>>[number];
//       ^ This Foo type has the property model ("Foo" string)
```
